drivers/lora/sx126x.c

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright (c) 2019 Manivannan Sadhasivam
+ * Copyright (c) 2020 Andreas Sandberg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/gpio.h>
+#include <drivers/lora.h>
+#include <drivers/spi.h>
+#include <zephyr.h>
+
+#include <sx126x/sx126x.h>
+
+#include "sx12xx_common.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(sx126x, CONFIG_LORA_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(semtech_sx1261)
+#define DT_DRV_COMPAT semtech_sx1261
+#define SX126X_DEVICE_ID SX1261
+#elif DT_HAS_COMPAT_STATUS_OKAY(semtech_sx1262)
+#define DT_DRV_COMPAT semtech_sx1262
+#define SX126X_DEVICE_ID SX1262
+#else
+#error No SX126x instance in device tree.
+#endif
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(semtech_sx1261) +
+	     DT_NUM_INST_STATUS_OKAY(semtech_sx1262) <= 1,
+	     "Multiple SX12xx instances in DT");
+
+#define HAVE_GPIO_CS		DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
+#define GPIO_CS_LABEL		DT_INST_SPI_DEV_CS_GPIOS_LABEL(0)
+#define GPIO_CS_PIN		DT_INST_SPI_DEV_CS_GPIOS_PIN(0)
+
+#define GPIO_RESET_LABEL	DT_INST_GPIO_LABEL(0, reset_gpios)
+#define GPIO_RESET_PIN		DT_INST_GPIO_PIN(0, reset_gpios)
+#define GPIO_RESET_FLAGS	DT_INST_GPIO_FLAGS(0, reset_gpios)
+
+#define GPIO_BUSY_LABEL		DT_INST_GPIO_LABEL(0, busy_gpios)
+#define GPIO_BUSY_PIN		DT_INST_GPIO_PIN(0, busy_gpios)
+#define GPIO_BUSY_FLAGS		DT_INST_GPIO_FLAGS(0, busy_gpios)
+
+#define GPIO_IRQ_LABEL		DT_INST_GPIO_LABEL(0, irq_gpios)
+#define GPIO_IRQ_PIN		DT_INST_GPIO_PIN(0, irq_gpios)
+#define GPIO_IRQ_FLAGS		DT_INST_GPIO_FLAGS(0, irq_gpios)
+
+#define HAVE_GPIO_ANTENNA_ENABLE			\
+	DT_INST_NODE_HAS_PROP(0, antenna_enable_gpios)
+#define GPIO_ANTENNA_ENABLE_LABEL			\
+	DT_INST_GPIO_LABEL(0, antenna_enable_gpios)
+#define GPIO_ANTENNA_ENABLE_PIN				\
+	DT_INST_GPIO_PIN(0, antenna_enable_gpios)
+#define GPIO_ANTENNA_ENABLE_FLAGS			\
+	DT_INST_GPIO_FLAGS(0, antenna_enable_gpios)
+
+#if DT_INST_NODE_HAS_PROP(0, dio2_tx_enable)
+#define DIO2_TX_ENABLE DT_INST_PROP(0, dio2_tx_enable)
```

Comment:
Boolean values (such as `DT_INST_PROP(0, dio2_tx_enable)`) are always set to 0  or 1, so there is no need to check with `DT_INST_NODE_HAS_PROP()`. @mbolivar-nordic can you confirm that?

Reply:
Ack. Tested without the conditional enable and it seems to behave as expected.